pkg/volume/csi/csi_plugin.go

```diff
@@ -334,8 +340,28 @@ func (p *csiPlugin) CanSupport(spec *volume.Spec) bool {
 	return spec.PersistentVolume != nil && spec.PersistentVolume.Spec.CSI != nil
 }
 
-func (p *csiPlugin) RequiresRemount() bool {
-	return false
+func (p *csiPlugin) RequiresRemount(spec *volume.Spec) bool {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.CSIServiceAccountToken) {
+		return false
+	}
+	kletHost, ok := p.host.(volume.KubeletVolumeHost)
+	if ok {
+		if err := kletHost.WaitForCacheSync(); err != nil {
```

Comment:
good point. it is actually will block forever. @msau42 this is a interface method that used in many places. i think we need to give a timeout in most cases.

Reply:
I think instead of calling `WaitForCacheSync` here, which gets called very frequently, it would be better to do this during volume manager initialization in kubelet. It looks like we already have plumbing to wait for all sources to be ready: https://github.com/kubernetes/kubernetes/blob/2ee1003430ea42060b48de9b28ecab2ce846c366/pkg/kubelet/volumemanager/volume_manager.go#L260

Reply:
done. cache will be syncing in the initialization. in a future PR, might clean up the places calling `WaitForCacheSync` as it is not necessary.